### PR TITLE
Fixes for Deep Earth Heating Pump

### PR DESCRIPTION
**Pull request by Nockyx**
After https://github.com/GTNewHorizons/GT5-Unofficial/pull/3379, while testing the changes in a playthrough i noticed several problems regarding the functionality of the Multiblock due to the old untouched code.

1. First reduced the EU comsuption from 512 EU/t to 480 EU/t to account for cable/transformer loss as other machines do.
2. Added Mode Changing button functionality to the GUI and rewrite of the mode code to match modern implementations.
3. Added Custom Operating texts to the GUI and splitted the method in MTEDrillerBase that was set as private for no reason
4. Fixed Bug where the multiblock automatically retracted pipe instead of working.

### CODE REVIEW

File: src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java [hunk 1]
```diff
@@ -234,18 +247,19 @@ protected boolean workingDownward(ItemStack aStack, int xDrill, int yDrill, int
...
                 this.consumeFluid(FluidRegistry.getFluid("ic2coolant"), coolantConverted);
                 this.addOutput(FluidRegistry.getFluidStack("ic2hotcoolant", (int) coolantConverted));
-            } else {
-                this.explodeMultiblock();
-                return false;
             }
+            // else {
+            // this.explodeMultiblock();
+            // return false;
+            // }
```
Comment (serenibyss) [inline src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java:262]: Is this meant to be commented out?

Comment (Nockyx) [reply src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java:262]: I forgot to uncomment while testing, my bad. It's not supposed to. I'll fix it

File: src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java [hunk 2]
```diff
@@ -297,7 +310,7 @@ private boolean consumeFluid(Fluid fluid, long ammount) {
     @Override
     protected void setElectricityStats() {
         try {
-            this.mEUt = this.isPickingPipes ? -60 : -1 * Math.toIntExact(GTValues.V[this.getMinTier()]);
+            this.mEUt = this.isPickingPipes ? -60 : (-480);
```
Comment (serenibyss) [inline src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java:313]:
```suggestion
            this.mEUt = this.isPickingPipes ? -60 : (-TierEU.RECIPE_HV);
```
Requires the import commented above

File: src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java [hunk 3]
```diff
@@ -22,27 +22,33 @@
...
 import net.minecraftforge.common.util.ForgeDirection;
 import net.minecraftforge.fluids.Fluid;
 import net.minecraftforge.fluids.FluidRegistry;
 
+import com.gtnewhorizons.modularui.api.math.Pos2d;
+
 import gregtech.api.enums.GTValues;
 import gregtech.api.enums.ItemList;
 import gregtech.api.enums.Materials;
+import gregtech.api.gui.modularui.GTUITextures;
```
Comment (serenibyss) [inline src/main/java/bartworks/common/tileentities/multis/MTEDeepEarthHeatingPump.java:35]:
```suggestion
import gregtech.api.enums.TierEU;
import gregtech.api.gui.modularui.GTUITextures;
```
Import so you can apply the change below in the UI

**Review by serenibyss — APPROVED** [review]

**Pull request merged**